### PR TITLE
[CALCITE-2882] Set dirty true when open connection

### PR DESCRIPTION
**Pull request by bakea**
Set dirty false in connectionPropertis in order to peoperties can be sync when createConnection again

### CODE REVIEW

**Comment by bakea** [comment]
Someone review it?

**Comment by F21** [comment]
@bakea Do you think it would be possible to add a test for this?

**Comment by joshelser** [comment]
I don't think it is by design that we are automatically re-creating a new connection for you after your previous connection was closed due to lack of use. What is triggering the re-connection that you refer to?

**Comment by zabetak** [comment]
I went through the JIRA case, I get what's the problem, but I don't get why and where it occurs. Without a test is difficult to verify that the fix is valid.

**Comment by bakea** [comment]
> I don't think it is by design that we are automatically re-creating a new connection for you after your previous connection was closed due to lack of use. What is triggering the re-connection that you refer to?

I find this in phoenix queryserver, i initialize a avaticaConnection which cache an phoenixConnection in the queryserver. If avaticaConnection idle exceed the phoenixConnection cache keepalive time, it will be reopen a phoenixConnection when i use it again. 
But the param dirty is false, connectionPropertis cannot be sync to phoenixconnection such as autocommit...

**Comment by bakea** [comment]
> I went through the JIRA case, I get what's the problem, but I don't get why and where it occurs. Without a test is difficult to verify that the fix is valid.

@zabetak  I add a test for this problem, if dirty set false after connProperties sync, although I set autocommit false, when connection reopen, autocommit is still effect，data still be queried.
If dirty initialize true when open connection everytime，query nothing when connection reopen.

File: server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java [hunk 1]
```diff
@@ -163,8 +165,12 @@ public String getJdbcUrl(int port, Serialization serialization, String urlSuffix
     static JdbcMeta getInstance() {
       if (instance == null) {
         try {
-          instance = new JdbcMeta(CONNECTION_SPEC.url, CONNECTION_SPEC.username,
-              CONNECTION_SPEC.password);
+          Properties info = new Properties();
+          info.put(JdbcMeta.ConnectionCacheSettings.EXPIRY_DURATION.key(), "10");
+          info.put(JdbcMeta.ConnectionCacheSettings.EXPIRY_UNIT.key(), TimeUnit.SECONDS.name());
```
Comment (danny0405) [inline server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java]: Does this expire time config has any side effect with the test already ? I'm a little worry about it, maybe you can make these 2 params configurable and set them in your test case.

File: server/src/test/java/org/apache/calcite/avatica/remote/RemoteMetaTest.java [hunk 2]
```diff
@@ -713,6 +713,40 @@ private int getSales(Connection conn, String salesTable, int productId) throws E
...
+
+      Statement statement = conn.createStatement();
+      final String create =
+              String.format(Locale.ROOT, "create table if not exists %s ("
+                      + "  id int not null, msg varchar(255) not null)", t);
+      int status = statement.executeUpdate(create);
+      assertEquals(status, 0);
+
+      Thread.sleep(10000);
+      statement = conn.createStatement();
```
Comment (danny0405) [inline server/src/test/java/org/apache/calcite/avatica/remote/RemoteMetaTest.java]: That really sleep too much time,  better to config this timeout params just in this case and make it fast enough, if it is much slower, maybe you should add this test case in [1].
[1] https://github.com/apache/calcite/blob/af97f55f1fcbb3cf6005393dffd91eb345c04b6f/core/src/test/java/org/apache/calcite/test/CalciteSuite.java#L19

File: server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java (hunk 1 above)
Comment (bakea) [reply server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java]: ok.I will do that.THX

File: server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java [hunk 3]
```diff
@@ -70,6 +71,31 @@ public void startServers() throws Exception {
     serversBySerialization.put(Serialization.PROTOBUF, protobufServer);
   }
 
+  /**
+   * Starts Avatica servers for each serialization type with the properties.
+   */
+  public void startServers(Properties properties) {
```
[the remark this answers is not in the document]
Comment (bakea) [reply server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java]: yeah! That looks more normalized.

File: server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java [hunk 4]
```diff
@@ -164,7 +190,20 @@ static JdbcMeta getInstance() {
...
+                  CONNECTION_SPEC.password);
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+      }
+      return instance;
+    }
+
+    static JdbcMeta getInstance(Properties properties) {
+      if (instance == null) {
```
[the remark this answers is not in the document]
Comment (bakea) [reply server/src/test/java/org/apache/calcite/avatica/remote/AvaticaServersForTest.java]: Good suggestion, I made a mistake. But I cannot find better solution, so I new a `JdbcMeta` instance in startServers method without `FullyRemoteJdbcMetaFactory`.

File: server/src/test/java/org/apache/calcite/avatica/remote/ConnectionPropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      assertFalse(remoteConn1.getAutoCommit());
+      assertEquals(remoteConn1.getTransactionIsolation(),
+              Connection.TRANSACTION_REPEATABLE_READ);
+    } finally {
+      ConnectionSpec.getDatabaseLock().unlock();
+    }
+  }
+
+  private static Connection getConnection(JdbcMeta m, String id) throws Exception {
+    Field f = JdbcMeta.class.getDeclaredField("connectionCache");
```
[the remark this answers is not in the document]
Comment (bakea) [reply server/src/test/java/org/apache/calcite/avatica/remote/ConnectionPropertiesTest.java]: The test case isnot under the same package with JdbcMeta, So I use refleaction.

**Pull request closed without merging**